### PR TITLE
Sc ar set

### PR DESCRIPTION
**Pull request by Scott-Canning**
Implemented set using `arweave.js`

### CODE REVIEW

File: src/setRecord.js [hunk 1]
```diff
@@ -1,31 +1,57 @@
...
     } catch (error) {
-        console.error(`Error in ${dryRun ? 'setRecord *dry run*' : 'setRecord'}: `, error);
+        console.error('Error in setRecord: ', error);
         return undefined;
     }
+}
+
+async function writeInteraction(arweave, input, contractId, privateKey) {
+    const interactionTx = await arweave.createTransaction({ 
+        data: Math.random().toString().slice(-4)
```
[the remark this answers is not in the document]
Comment (aboutlo) [reply src/setRecord.js:42]: Why `interaction` ?

A better name could be `postTransaction(data, contractAddress)`
but then we have to deal with arweave and the private key so we could use a closure and do currying.


we could do `createClient(arweave, privatekey, dryRun).setRecord(antAddress, manifestId)`

createClient can be extracted as function used between `setRecord` and `upload`


so an idea of the implementation could be:

import {setRecord, upload}

function createClient({arweave, privatekey, dryRun = false}) {
  return {
      setRecord(arweave, privateKey, dryRun) 
      upload(),
  }
}

... 

function setRecord ({arweave, privateKey, dryRun}) {
   return () => {
     // do your stuff here
    const tx  = buildTransaction(arweave,privateKey)  
     await arweave.transactions.sign(tx, privateKey);
     const response = await arweave.transactions.post(tx);
     return response
   }
}

Anyway, let's do this refactoring AFTER we merge this

File: src/setRecord.test.js [hunk 2]
```diff
@@ -1,60 +1,50 @@
...
+        assert.strictEqual(response.statusText, 'OK');
+        assert.strictEqual(transactionId, 'dry-run-tx-id')
     });
 
     it('should successfully returns a write interaction response', async () => {
-        const response = await setRecord('validAntAddress', 'validManifestId', privateKeyMock, false);
-        assert.ok(response);
-        assert.strictEqual(response.state.owner, 'ownerAddress');
-        assert.strictEqual(response.state.records['@'].transactionId, 'validManifestId');
+        const { response, transactionId } = await setRecord('validAntAddress', 'validManifestId', privateKeyMock, false);
```
Comment (aboutlo) [inline src/setRecord.test.js:45]: here you should check that the mocked methods have been called correctly by setRecord 

e.g. from upload.test.js
```
    assert.equal(getStatusMock.mock.callCount(), 2)
    assert.equal(getUploaderMock.mock.callCount(), 1)
    assert.deepEqual(getStatusMock.mock.calls[0].arguments, [txId])
    assert.deepEqual(getStatusMock.mock.calls[1].arguments, [txId])
```

**Review by aboutlo — CHANGES_REQUESTED** [review]
A couple of changes and then we are good to go

File: src/setRecord.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Scott-Canning) [reply src/setRecord.js:42]: the `writeInteraction` naming convention is consistent with the modern `warp-contracts` Arweave SmartWeave SDK used specifically for making SmartWeave contract interactions (SmartWeaveActions). See: https://academy.warp.cc/docs/sdk/basic/contract-methods. `postTransaction` may be too narrow in that the function encapsulates creating, signing, and posting SmartWeaveActions

**Review by aboutlo — APPROVED** [review]
LGTM

**Pull request merged**